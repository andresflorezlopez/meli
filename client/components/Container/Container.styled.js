import styled from "styled-components";

export const StyledContainer = styled.div`
  flex-direction: ${({ direction }) => direction};
  max-width: 1200px;
  margin: 0 auto;
  display: flex;
  ${({ theme }) => theme.breakpoints.md} {
    width: 95%;
  }
`;

export default StyledContainer;
