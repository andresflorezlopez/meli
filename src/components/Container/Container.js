import React from "react";
import PropTypes from "prop-types";

import StyledContainer from "./Container.styled";

const Container = ({ children, direction }) => {
  return <StyledContainer direction={direction}>{children}</StyledContainer>;
};

Container.propTypes = {
  children: PropTypes.node.isRequired,
  direction: PropTypes.oneOf(["row", "column"])
};

Container.defaultProps = {
  direction: "column"
};

export default Container;
