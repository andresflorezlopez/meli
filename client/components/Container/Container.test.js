import React from "react";
import Container from "./Container";
import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("Container", () => {
  test("should render", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <Container>
          <p>Magic container</p>
        </Container>
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
