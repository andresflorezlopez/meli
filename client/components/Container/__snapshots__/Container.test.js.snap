// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render 1`] = `
<div
  className="sc-1ctswtc-0 dyKYyT"
  direction="column"
>
  <p>
    Magic container
  </p>
</div>
`;
